### DISCUSSION TITLE
Websocket cannot trace client's IP?

### DISCUSSION THREAD

**Discussion by Zhaoxun** [fastapi/fastapi#11648] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
################utils.py##########################################
import logging
logger = logging.getLogger("test")
sh = logging.StreamHandler()
sh.setLevel(logging.INFO)
sfmt = logging.Formatter(
        "%(asctime)s - %(levelname)s [%(filename)s:%(lineno)d] - %(message)s",
)
sh.setFormatter(sfmt)
logger.addHandler(sh)

log = logger

############main.py#############

from fastapi import FastAPI, Request
from fastapi import WebSocket, WebSocketDisconnect
from utils import log
from users import ws

WS = FastAPI()

@WS.websocket('/')
async def host_client(request:Request, session: WebSocket):
    log.warning('from', request.client.host)
......
```

### Description

I cannot retrieve client's IP like in the http connection/request, just like in [this page](https://www.iditect.com/faq/python/fastapi-starlette-get-client-real-ip.html)

### Operating System

Linux

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

3.9.16

**Comment by YuriiMotov** [comment, marked as answer]
It can be done this way:

```python
from fastapi import FastAPI
from fastapi import WebSocket

WS = FastAPI()

@WS.websocket('/')
async def host_client(websocket: WebSocket):
    print('from', websocket.client.host)

```
